### ISSUE TITLE
Bug report - [Stuck at "Starting service"]

### ISSUE THREAD

**Issue by lene** [CypherpunkArmory/UserLAnd#1013]
I cannot get UserLAnd to run any app on my Lenovo Yoga Book under Android 7.1.1. It runs fine on my HTC U11 Life with Android 9.

# Describe the bug

After selecting an app, UserLAnd downloads and extracts assets, then is stuck with the message "Starting service" forever. No terminal session is opened.

# Steps to reproduce the behavior:

* Install UserLand
* Open UserLand
* In the "Apps" screen, click on any app

# Screenshots

N/A

# Device Information

Device: Lenovo Yoga Book
Android Version: Android 7.1.1
UserLAnd Version: Can't find a version number. Installed version is published Aug 12, 2019  
Whether the device is rooted: No
Whether the device is running LineageOS: No

Please add any other context you think might help us resolve your issue:

As mentioned above, UserLAnd seems to work fine on my HTC U11 Life under Android 9.

I have checked issues #71, #235 and #567 and the suggested solutions/workarounds did not help.

Two sample PRoot files, trying to start Ubuntu and Alpine, seem to show me that it seems to start a 64bit bash binary on a 32bit system.
[Proot_Debug_Log_ubuntu.txt](https://github.com/CypherpunkArmory/UserLAnd/files/3513334/Proot_Debug_Log_ubuntu.txt)
[Proot_Debug_Log_alpine.txt](https://github.com/CypherpunkArmory/UserLAnd/files/3513335/Proot_Debug_Log_alpine.txt)

**Comment by lene** [comment]
Looking through the issues with different search terms, I think this might be a duplicate of #977

**Comment by phone-burner** [comment]
Me too.You can try To reboot your phone and then start UserLAnd.

**Comment by lene** [comment]
Rebooting does not work, neither does deleting all caches, uninstalling, rebooting and then reinstalling UserLAnd (as recommended in #235)

**Comment by EoflaOE** [comment]
Same thing happened in my Lenovo TB3-710F Android 5.0.1.

**Comment by phone-burner** [comment]
I found that UserLAnd 2.6.2 doesn't kill the daemons started by itself if it's not shutdown properly(type in 'logout' in the terminal and then click 'stop service'),neither does Android.As a result,you cannot start Ubuntu,Kali or Debian anymore on your phone unless you reboot your phone.These daemons cannot be shown or controlled by Android.If you have this problem,just remember to shut down UserLAnd properly.

**Comment by EoflaOE** [comment]
Fixed on latest beta version of UserLAnd using Google Play Store and it now works , version 2.6.2-beta-663767

**Comment by MatthewTighe** [comment]
Providing a [PRoot Debug Log](https://github.com/CypherpunkArmory/UserLAnd/wiki/Generating-PRoot-Debug-Logs) would likely be helpful in resolving this issue.

**Comment by lene** [comment]
What's wrong with the two debug logs I attached to the issue?

**Comment by MatthewTighe** [comment]
My eyesight, apparently, Sorry about that!

**Comment by mesprylum** [comment]
I think I'm experiencing this (motorola moto e5, android 8.0.0). I had been using octave. It isn't obvious to me why, but I got this problem after my phone updated (to Android security patch level 1/7/2019, I think). Or it could be unrelated and more that the update forced me to restart the app. It had been working well enough, but now I just get "Starting service..." and nothing else happens, as the OP says.

Restarting the phone doesn't help, in my case. Nor does the latest (or latest as of last Friday) beta version as suggested above.

**Comment by corbinlc** [comment]
@lene  v2.6.3 was just released (may take a few hours until you see the option to update).  It fixes many issues, but one of the issues it _should_ fix is with the Lenovo Yoga Books.  We started bundling certain libraries with the app instead of downloading everything and the app installer on that device was installing a different version of the libraries than what the devices build information stated.  We are now matching the filesystem type to what version of the libraries your device installs, as opposed to trusting the build information.  This should fix this issue.

Others, I don't know if you are having the same issue or not.  If v2.6.3 does not help you, please file new issues with more info.

**Comment by ONLY-F4-YOU** [comment]
In 2023 I also fetch the problem. I'm using Poco c31,non-rooted, android version 10. How can I fix this issue 🙏

**Comment by krasnit** [comment]
Same issue on Moto G pure 2023: Stuck at "Starting Service" when trying to run any distro in graphical mode.  Works only in terminal mode.  Android 12.  Does not generate PRoot_Debug_Log

**Comment by milen-prg** [comment]
This BUG persist in last days of October 2024, so seems from the last several years, this app UserLAnd can't be fixes and can't work.
Unfortunately until now there is no Linux possibility for standard Android phone (as Termux stucked on signal 9 problem forever). ☹️
